### PR TITLE
fix: align validations with JS

### PR DESCRIPTION
**Pull request by lizaiv77**
- add methods to validate for mandatory and unsupported properties
- add validations for ^
- add validations for empty strings where it was missing
- error codes changes due to using more generic codes and skipping gaps

### CODE REVIEW

File: utam-compiler/src/main/resources/errors.config.json [hunk 1]
```diff
@@ -81,18 +89,10 @@
...
   {
     "code": 204,
-    "message": "element \"%s\": property \"%s\" is mandatory"
-  },
-  {
-    "code": 205,
-    "message": "element \"%s\": %s element can't have nested elements"
-  },
-  {
-    "code": 206,
```
Comment (lizaiv77) [inline utam-compiler/src/main/resources/errors.config.json:95]: replaced with generic error codes 8 and 9

File: utam-compiler/src/main/resources/errors.config.json [hunk 2]
```diff
@@ -135,6 +130,10 @@
     "message": "method \"%s\": incorrect format of compose method",
     "docs": "https://utam.dev/grammar/spec#compose-method"
   },
+  {
+    "code": 501,
+    "message": "%s: parameter \"%s\" at method level can't have type \"argumentReference\""
+  },
```
Comment (lizaiv77) [inline utam-compiler/src/main/resources/errors.config.json:136]: instead 504 to not skip code

File: utam-compiler/src/main/resources/errors.config.json [hunk 3]
```diff
@@ -108,11 +108,6 @@
     "code": 302,
     "message": "element \"%s\" filter: filter can only be set for list"
   },
-  {
-    "code": 303,
-    "message": "method \"%s\": incorrect format of statement filter",
-    "docs": "https://utam.dev/grammar/spec#element-filter"
-  },
```
Comment (lizaiv77) [inline utam-compiler/src/main/resources/errors.config.json:115]: replaced by 1200 and 1201

File: utam-compiler/src/main/resources/errors.config.json [hunk 4]
```diff
@@ -202,8 +201,8 @@
     "message": "method \"%s\" statement: incorrect return type; expected \"%s\", provided is \"%s\""
   },
   {
-    "code": 614,
-    "message": "method \"%s\" statement: expected type for matcher is \"%s\", found \"%s\""
+    "code" : 614,
+    "message": "method \"%s\" statement: to use chain previous statement should return custom type, but it returns \"%s\""
```
Comment (lizaiv77) [inline utam-compiler/src/main/resources/errors.config.json:205]: instead 617

File: utam-compiler/src/main/resources/errors.config.json [hunk 5]
```diff
@@ -181,6 +176,10 @@
     "code": 607,
     "message": "method \"beforeLoad\" statement: only \"document\" or \"root\" elements are allowed"
   },
+  {
+    "code" : 608,
+    "message": "method \"%s\" statement: \"returnSelf\" can only be invoked from a last statement"
+  },
```
Comment (lizaiv77) [inline utam-compiler/src/main/resources/errors.config.json:182]: instead 618

File: utam-compiler/src/main/resources/errors.config.json [hunk 6]
```diff
@@ -271,16 +267,12 @@
   },
   {
     "code": 904,
-    "message": "property \"%s\" is not supported, interface declaration can only have properties: [ %s ]"
-  },
-  {
-    "code": 905,
```
Comment (lizaiv77) [inline utam-compiler/src/main/resources/errors.config.json:277]: use 8 and 9 instead

**Review by jimevanssfdc — APPROVED** [review]
LGTM. We should document the error codes outside the code bases themselves in a document (either in Quip or at utam.dev).

**Pull request merged**